Pass credentials as parameters

rjvandoesburg opened the pull request:
These changes will pass the credentials as parameters at the creation of a new MongoDB connection instead of passing them on the DSN.

reference: https://github.com/jenssegers/laravel-mongodb/pull/748

**Username and password now need to be in the options section of the DB config**

jenssegers commented:
I would prefer having the username and password outside of the `options` in the Laravel configuration (like it used to be). And pass them to the `options` in the `Connection` class.

rjvandoesburg commented:
Ok, I will change that back and add the credentials to the constructor in the `Connection` class

rjvandoesburg commented:
Ok, so the changes I've made are as followed.
In the `Connection` class I've added a couple of checks to see if a person has already provided credentials in the options, if not the credentials will be used from the config (the old way)

I've reverted the readme as no changes are there and I've changed the access level of a method because it was different from the access level of the method it was overriding.

But it seems I've managed to mess up the tests... I'll be fixing those next :)

jenssegers commented on src/Jenssegers/Mongodb/Connection.php:
```diff
@@ -136,6 +136,14 @@ protected function createConnection($dsn, array $config, array $options)
         }
 
+        // Check if the credentials are not already set in the options
+        if (!isset($options['username'])) {
```
Can you turn this into:

```
if (! isset($options['username']) && isset($config['username'])) {
$options['username'] = $config['username'];
```

rjvandoesburg replied:
Yes, I've seen that that is indeed where the problem was.

I've got the test suite running locally now as well so I can test before pushing bad code again.
I want to test the entire suite before pushing so I just need to get MySQL up and running, after that I will submit the new changes again

rjvandoesburg commented:
Tests are working again. Applied the change you found as well.

jenssegers commented:
Can you squash your commits?

rjvandoesburg commented:
Ah sorry missed the notification.
Cheers for accepting the PR

oneso commented:
You are not checking if the credentials are empty. This resulty in an "Authentication failed." exception.
Two options:

- add empty() check
or
- edit readme:
remove 'DB_USERNAME/DB_PASSWORD' from .env if empty and change database.php from:
```
env($test . 'DB_USERNAME', '')
env($test . 'DB_PASSWORD', '')
```
to:
```
env($test . 'DB_USERNAME')
env($test . 'DB_PASSWORD')
```

Or is it just me?

jenssegers commented:
I added it back again.

rjvandoesburg commented:
Hmm, I see what you mean.
However I do think there is a difference in having empty fields or no fields at all.
I'd suggest changing the readme so it states to remove the username and password if you wish to use anonymous login.

By having empty username and password a user might be alerted for misconfiguration in the .env file.
